mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -1741,9 +1742,15 @@ void mlir::affine::getComputationSliceState(
     //    2. Loop bounds of the source and destination match.
     //    3. Is being inserted at the innermost insertion point.
     std::optional<bool> isMaximal = sliceState->isMaximal();
-    if (isLoopParallelAndContainsReduction(getSliceLoop(i)) &&
-        isInnermostInsertion() && srcIsUnitSlice() && isMaximal && *isMaximal)
-      continue;
+    // Skip paralellism check of sibling if maximal fusion is true.
+    if (!fusionstrategy.isMaximalFusion()) {
+      if (isLoopParallelAndContainsReduction(getSliceLoop(i)) &&
+          isInnermostInsertion() && srcIsUnitSlice() && isMaximal && *isMaximal)
```

Comment:
For eg., one can add an option `skipParallelismCheck`. 

Lower-level utilities shouldn't refer to or encode any semantics from higher-level/client utilities.